Code-Hammers/code-hammers
# [CHE-155] Create PostgreSQL Dev Environment

### Description

The PR was originally intended to set up the docker dev environment postgres container and ensure connectivity to the app as part of the [CHE-140](https://codehammers.atlassian.net/browse/CHE-140?atlOrigin=eyJpIjoiNDZkNDZmYTYxNjkzNGZlZGI3YTEzODg0ODc3MWJiN2YiLCJwIjoiaiJ9) epic to create our job applications tracker. I went 'slightly' out of scope on this and also:

1. Created a script to seed database with sample applications
2. Created Router and a basic endpoint to fetch all applications
3. Created an Applications page for front end and hooked up Router navigation
4. Created a basic fetch and display from front end to confirm full stack connection

### Jira Task

[CHE-155](https://codehammers.atlassian.net/browse/CHE-155?atlOrigin=eyJpIjoiOThkNTljYzQ1MTdkNGZiYjllOGFhODZiMDRjNGU1ZDAiLCJwIjoiaiJ9)

### Testing Instructions

1. Ensure branches are all up to date
2. Use npm run docker-remove-all nd then confirm in Docker desktop all container, images and volumes have been removed.
3. Use nom run docker-dev from this feature branch
4. Log into application and make sure that there are no errors and that the Applications page loads and displays the seed data as expected
5. ***You will need to execute Step 2 and 3 again when you return to your dev branch to make sure you are using the correct images for your branch***

### Checklist


### Additional Notes, Images, etc.

![Screenshot 2024-05-16 at 9 18 27 AM](https://github.com/Code-Hammers/code-hammers/assets/62915047/bd61895c-995c-4fd2-964e-0dfdc6a3afcb)



[CHE-140]: https://codehammers.atlassian.net/browse/CHE-140?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CHE-155]: https://codehammers.atlassian.net/browse/CHE-155?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
